packages/netlify-cms-backend-git-gateway/src/implementation.ts

```diff
@@ -193,53 +255,52 @@ export default class GitGateway {
     return user && user.logout();
   }
   getToken() {
-    return this.tokenPromise();
+    return this.tokenPromise!();
   }
 
-  entriesByFolder(collection, extension) {
-    return this.backend.entriesByFolder(collection, extension);
-  }
-  entriesByFiles(collection) {
-    return this.backend.entriesByFiles(collection);
+  entriesByFolder(folder: string, extension: string, depth: number) {
```

Comment:
Each backend had the same code to extract the folder and depth from the collection so I moved that to the core